lib/util/http/index.ts

```diff
@@ -91,7 +91,7 @@ export class Http<GetOptions = HttpOptions, PostOptions = HttpPostOptions> {
       url = resolveBaseUrl(httpOptions.baseUrl, url);
     }
 
-    let options: GotOptions = merge<GotOptions>(
+    let options: HttpOptions['gotOptions'] = merge<HttpOptions['gotOptions']>(
```

Comment:
```suggestion
    let options: GotOptions = merge<GotOptions>(
```

Again, use interfaces directly.